cishouseholds/merge.py

```diff
@@ -40,10 +47,22 @@ def union_multiple_tables(tables: List[DataFrame]):
         list of objects representing the respective input tables
     """
     merged_df = tables[0]
-
     for i, table_n in enumerate(tables[1:]):
-        merged_df, dfn = prepare_for_union(merged_df, table_n)
-        merged_df = merged_df.unionByName(dfn)
+        merged_df, dfn = prepare_for_union(merged_df, table_n, combine_order=reorder_columns)
```

Comment:
I think looping `prepare_for_union` between the pairs of tables is slowing this down. Please could you try creating a set of unique columns from the list of dataframes, then update each dataframe to have the complete set.

If possible, we want to keep the table-writing code separate from the other logic so that it can be effectively tested.